File: threads/src/main/java/im/threads/ui/fragments/BaseFragment.java
package im.threads.ui.fragments;

import androidx.fragment.app.Fragment;

import im.threads.ui.config.Config;
import im.threads.ui.utils.ToastUtils;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public abstract class BaseFragment extends Fragment {

    private CompositeDisposable compositeDisposable;

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unsubscribeAll();
    }

    protected boolean subscribe(final Disposable event) {
        if (compositeDisposable == null || compositeDisposable.isDisposed()) {
            compositeDisposable = new CompositeDisposable();
        }
        return compositeDisposable.add(event);
    }

    private void unsubscribeAll() {
        if (compositeDisposable != null) {
            compositeDisposable.dispose();
            compositeDisposable = null;
        }
    }

    public void showToast(final String message) {
        if (Config.getInstance().getChatStyle().isToastStylable())
            ToastUtils.showSnackbar(getContext(), getView().getRootView(), message);
        else {
            ToastUtils.showToast(getContext(), message);
        }
    }
}

File: src/main/java/im/threads/ui/fragments/BaseFragment.kt
package im.threads.ui.fragments

import androidx.fragment.app.Fragment
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

abstract class BaseFragment() : Fragment() {
    private var compositeDisposable: CompositeDisposable? = null
    override fun onDestroyView() {
        super.onDestroyView()
        unsubscribeAll()
    }

    protected fun subscribe(event: Disposable?): Boolean {
        if (compositeDisposable == null || compositeDisposable?.isDisposed == true) {
            compositeDisposable = CompositeDisposable()
        }
        return if (event != null) {
            compositeDisposable?.add(event) ?: false
        } else {
            false
        }
    }

    private fun unsubscribeAll() {
        if (compositeDisposable != null) {
            compositeDisposable?.dispose()
            compositeDisposable = null
        }
    }
}
